tests/integration/conftest.py

```diff
@@ -1,9 +1,42 @@
 import pytest
+from eth.exceptions import HeaderNotFound
 
-from ape import networks
+from ape import chain, networks
+from ape.exceptions import ChainError
 
 
-@pytest.fixture
-def eth_tester_provider():
+@pytest.hookimpl(trylast=True, hookwrapper=True)
+def pytest_collection_finish(session):
     with networks.parse_network_choice("::test"):
-        yield networks.active_provider
+        yield
+
+
+@pytest.hookimpl(hookwrapper=True)
+def pytest_runtest_protocol(item, nextitem):
+    snapshot_id = chain.snapshot()
+    yield
+
+    try:
+        chain.restore(snapshot_id)
+    except (HeaderNotFound, ChainError):
```

Comment:
This seems like a bug and I am hoping to figure this out more and remove this part...

The tests work locally but I am having troubles in CI.